auth.ts

```diff
@@ -1,35 +1,10 @@
-import NextAuth, { type DefaultSession } from 'next-auth'
-import GitHub from 'next-auth/providers/github'
-import { NextResponse } from 'next/server'
+import { createServerActionClient } from '@supabase/auth-helpers-nextjs'
+import { cookies } from 'next/headers'
 
-declare module 'next-auth' {
-  interface Session {
-    user: {
-      /** The user's id. */
-      id: string
-    } & DefaultSession['user']
-  }
+export const auth = async () => {
+  // Create a Supabase client configured to use cookies
+  const supabase = createServerActionClient({ cookies })
+  const { data, error } = await supabase.auth.getSession()
+  if (error) throw error
+  return data.session
 }
-
-export const {
-  handlers: { GET, POST },
-  auth,
-  CSRF_experimental
-} = NextAuth({
-  providers: [GitHub],
-  callbacks: {
-    jwt({ token, profile }) {
-      if (profile) {
-        token.id = profile.id
-        token.image = profile.picture
-      }
-      return token
-    },
-    authorized({ auth }) {
-      return !!auth?.user
-    }
-  },
-  pages: {
-    signIn: '/sign-in'
-  }
```

Comment:
just to call out a potential issue - these are actual config/UX choices https://github.com/vercel-labs/ai-chatbot/pull/90/files i think you have mostly addressed them (except for the forced signin?) but just proactively calling it out

Reply:
Ah interesting, their deployed demo actually allows anonymous conversations without requiring you to log in. But yah, happy to add a middleware based redirect if user isn't logged in, is that your preference?

Reply:
Okay, added here with an comment that it's optional: https://github.com/thorwebdev/vercel-ai-chatbot/pull/1/commits/842803c979dddfb621a26eb55742cc4ce9806e26